src/OrchardCore.Modules/OrchardCore.AuditTrail/Startup.cs

```diff
@@ -88,7 +89,57 @@ public override void ConfigureServices(IServiceCollection services)
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderBy(i => i.CreatedUtc))
+                    .WithSelectListItem<Startup>((S, opt, model) => new SelectListItem(S["Oldest"], opt.Value, model.Sort == opt.Value));
+
+                options.ForSort("category-asc-time-desc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderBy(i => i.Category).ThenByDescending(i => i.CreatedUtc))
+                    .WithSelectListItem<Startup>((S, opt, model) => new SelectListItem(S["Category"], opt.Value, model.Sort == opt.Value));
+
+                options.ForSort("category-asc-time-asc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderBy(i => i.Category).ThenBy(i => i.CreatedUtc));
+
+                options.ForSort("category-desc-time-desc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderByDescending(i => i.Category).ThenByDescending(i => i.CreatedUtc));
+
+                options.ForSort("category-desc-time-asc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderByDescending(i => i.Category).ThenBy(i => i.CreatedUtc));
+
+                options.ForSort("event-asc-time-desc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderBy(i => i.Name).ThenByDescending(i => i.CreatedUtc))
+                    .WithSelectListItem<Startup>((S, opt, model) => new SelectListItem(S["Event"], opt.Value, model.Sort == opt.Value));
+
+                options.ForSort("event-asc-time-asc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderBy(i => i.Name).ThenBy(i => i.CreatedUtc));
+
+                options.ForSort("event-desc-time-desc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderByDescending(i => i.Name).ThenByDescending(i => i.CreatedUtc));
+
+                options.ForSort("event-desc-time-asc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderByDescending(i => i.Name).ThenBy(i => i.CreatedUtc));
+
+                options.ForSort("user-asc-time-asc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderBy(i => i.NormalizedUserName).ThenBy(i => i.CreatedUtc))
+                    .WithSelectListItem<Startup>((S, opt, model) => new SelectListItem(S["User"], opt.Value, model.Sort == opt.Value));
+
+                options.ForSort("user-desc-time-desc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderByDescending(i => i.NormalizedUserName).ThenByDescending(i => i.CreatedUtc));
+
+                options.ForSort("user-desc-time-asc")
+                    .WithQuery((val, query) => query.With<AuditTrailEventIndex>().OrderByDescending(i => i.NormalizedUserName).ThenBy(i => i.CreatedUtc));
+
+                options.ForSort("user-desc-time-desc")
```

Comment:
@deanmarcussen looks like 'user-desc-time-desc' has been duplicated here. It throws an exception in the extension class in AuditTrailAdminOptions.cs.